addons/GearSwap/libs/closetCleaner.lua

```diff
@@ -0,0 +1,379 @@
+	else	
+		print('lua file for '..file..' not found!')
+	end 
+end
+
+-- this function tallies the items used in each lua file
+function cc.list_sets(t, f)  
+	write_sets = T{}
+	local print_r_cache={}
+    local function sub_print_r(t,fromTab)
+		if (type(t)=="table") then
+			for pos,val in pairs(t) do
+				if S{"WAR", "MNK", "WHM", "BLM", "RDM", "THF", "PLD", "DRK", "BST", "BRD", "RNG", "SAM", "NIN", "DRG", "SMN", "BLU", "COR", "PUP", "DNC", "SCH", "GEO", "RUN"}:contains(pos) then
+					job = pos
+				end
+				if (type(val)=="table") then
+					sub_print_r(val,job)
+				elseif (type(val)=="string") then
+					if val ~= "" and val ~= "empty" then 
+						if S{"name", "main", "sub", "range", "ammo", "head", "neck", "left_ear", "right_ear", "body", "hands", "left_ring", "right_ring", "back", "waist", "legs", "feet", "ear1", "ear2", "ring1", "ring2", "lear", "rear", "lring", "rring"}:contains(pos) then
+							if cc.sandbox.itemsByName[val:lower()] ~= nil then
+								itemid = cc.sandbox.itemsByName[val:lower()]
+							elseif cc.sandbox.itemsBylongName[val:lower()] ~= nil then
+								itemid = cc.sandbox.itemsBylongName[val:lower()]
+							else
+								print("Item: '"..val.."' not found in gearswap.resources! "..pos)
+							end
+							if write_sets[itemid] == nil then
+								write_sets[itemid] = 1
+								if cc.job_used[itemid] == nil then
+									-- print("found item "..itemid)
+									cc.job_used[itemid] = job
+									cc.job_logged[itemid..job] = 1
+								else
+									cc.job_used[itemid] = cc.job_used[itemid]..","..job
+									cc.job_logged[itemid..job] = 1
+								end
+							else	
+								write_sets[itemid] = write_sets[itemid] + 1
+								if cc.job_logged[itemid..job] == nil then
+									cc.job_used[itemid] = cc.job_used[itemid]..","..job
+									cc.job_logged[itemid..job] = 1
+								end
+							end
+						end
+					end
+				else
+					print("Error: Val needs to be table or string "..type(val))
+				end
+			end
+		end
+    end
+    sub_print_r(t,nil)
+	if ccDebug then
+		data = T{"Name", " | ", "Count", " | ", "Jobs", " | ", "Long Name"}
+		form = T{"%22s", "%3s", "%10s", "%3s", "%88s", "%3s", "%60s"}
+		cc.print_row(f, data, form)
+		cc.print_break(f, form)
+		f:write('\n')
+		for k,v in pairs(write_sets) do
+			data = T{gearswap.res.items[k].en, " | ", tostring(v), " | ", cc.job_used[k], " | ", gearswap.res.items[k].enl}
+			cc.print_row(f, data, form)
+			cc.sandbox.gsGear[k] = v
+		end
+		f:write()
+	else
+		for k,v in pairs(write_sets) do
+			cc.sandbox.gsGear[k] = v
+		end
+	end
+end
+
+-- interate throught table in a sorted order.
+function cc.spairs(t, order)
+    -- collect the keys
+    local keys = {}
+    for k in pairs(t) do keys[#keys+1] = k end
+
+    -- if order function given, sort by it by passing the table and keys a, b,
+    -- otherwise just sort the keys 
+    if order then
+        table.sort(keys, function(a,b) return order(t, a, b) end)
+    else
+        table.sort(keys)
+    end

```

Comment:
We do not have a sorted iterator, but the above few lines could be replaced by our library functions:
```lua
local keys = t:keyset():sort()
```

Or, if `t` is not a `T{}`-table:
```lua
local keys = table.sort(table.keyset(t))
```